tsdb/querier/querier.go

```diff
@@ -134,7 +134,7 @@ func (q *Query) exec(blockStream []tsdb.Block, encoding bool) interface{} {
 		// A nil range represents to return all time-series value as response.
 		if q.Range == nil {
 			base = QueryResponse{
-				TimeInvolved: time.Since(stamp),
+				TimeInvolved: time.Since(stamp).Nanoseconds(),
```

Comment:
I wouldn't do this here. The way this should be done is, at line 103, we must use a defer so that the only statement works for all cases and we make sure that there are no edge cases.

```golang
defer func() {
	q.queryResponse.TimeInvolved = time.Since(stamp).Nanoseconds()
}()
```

Reply:
And remove all other places for this calculation, as this itself is enough.

Note: Query time is the time from when you enter this func to exit of this func, since that's where the actual querying takes place. All other times fall under fetch time.

Reply:
In order to save from computations in UI, How about using `time.Since(stamp).String()`?

Reply:
- Absolutely right, `defer` will be best for the time duration calculation here. I will implement it that way.
- Yes, the `time.Since(stamp).String()` will help us avoid the calculations in the frontend, also then the return type will be a string I assume. I had a question, sending time as a string will prevent us from doing any calculation on the time in the frontend, that shouldn't be a problem, right?

Reply:
I don't think there are any calculations involved with time in the frontend, so this should be fine.